### PR TITLE
[Examples] 2.0 Collection/Table API updates

### PR DESCRIPTION
**Pull request by hannahmbanana**
Update examples to conform to the API changes in PR #2372. @Adlai-Holler: ready for review.

@maicki: could you look at the `CatDealsCollectionView` layout? I'm not able to figure out the sizing bug.

### CODE REVIEW

**Comment by hannahmbanana** [comment]
@Adlai-Holler: could you review & merge once Travis passes?

File: examples/AsyncDisplayKitOverview/Sample/OverviewComponentsViewController.h [hunk 1]
```diff
@@ -17,7 +17,7 @@
 //  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
-#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
```
Comment (appleguy) [inline examples/AsyncDisplayKitOverview/Sample/OverviewComponentsViewController.h]: This line says it all.  haha! <3

File: examples/AsyncDisplayKitOverview/Sample/OverviewComponentsViewController.m [hunk 2]
```diff
@@ -109,18 +134,17 @@ - (void)viewDidLoad
...
-    [self setupTableNode];
 }
 
 - (void)viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
     
-    ASTableView *tableView = self.tableNode.view;
-    [tableView deselectRowAtIndexPath:tableView.indexPathForSelectedRow animated:YES];
+    [_tableNode.view deselectRowAtIndexPath:_tableNode.view.indexPathForSelectedRow animated:YES];
```
Comment (appleguy) [inline examples/AsyncDisplayKitOverview/Sample/OverviewComponentsViewController.m]: Can we check if there is a tableNode version of this method?  Can you call deselect directly on the tableNode?

**Comment by hannahmbanana** [comment]
@Adlai-Holler

**Comment by Adlai-Holler** [comment]
I just landed #2453 which adds the selection API to the node versions of table/collection. @hannahmbanana would you be willing to update this diff to use that method?

**Pull request merged**